src/OpenTelemetry.Extensions.Enrichment/TraceEnricher.cs

```diff
@@ -0,0 +1,27 @@
+// <copyright file="TraceEnricher.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System;
+
+namespace OpenTelemetry.Extensions.Enrichment;
+
+public abstract class TraceEnricher : BaseEnricher<TraceEnrichmentBag>
+{
```

Comment:
nit: Add `protected TraceEnricher() {}` it is a [best-practice thing for abstract class to not have a public ctor](https://learn.microsoft.com/en-us/dotnet/standard/design-guidelines/abstract-class).

Reply:
Same comment for `BaseEnricher<T>` 😄

Reply:
Done